test/e2e-oidc/external_oidc_test.go

```diff
@@ -152,12 +152,22 @@ func TestExternalOIDCWithKeycloak(t *testing.T) {
 			{"invalid issuer URL", func(s *configv1.AuthenticationSpec) {
 				s.OIDCProviders[0].Issuer.URL = "https://invalid-idp.testing"
 			}},
-			// TODO: currently, this crashes the KAS pods during rollout
-			// enable this test once we've added CAO-side validations
-			// to prevent the crash; see: https://issues.redhat.com/browse/CNTRLPLANE-811
-			// {"empty username claim", func(s *configv1.AuthenticationSpec) {
-			// 	s.OIDCProviders[0].ClaimMappings.Username.Claim = ""
-			// }},
+			{"empty username claim", func(s *configv1.AuthenticationSpec) {
+				s.OIDCProviders[0].ClaimMappings.Username.Claim = ""
+			}},
+			{"uncompilable CEL expression for uid claim mapping", func(s *configv1.AuthenticationSpec) {
+				s.OIDCProviders[0].ClaimMappings.UID = &configv1.TokenClaimOrExpressionMapping{
+					Expression: "^&*!@#^*(",
+				}
+			}},
+			{"uncompilable CEL expression for extras claim mapping", func(s *configv1.AuthenticationSpec) {
+				s.OIDCProviders[0].ClaimMappings.Extra = []configv1.ExtraMapping{
+					{
+						Key:             "testing/key",
+						ValueExpression: "^&*!@#^*(",
+					},
+				}
+			}},
```

Comment:
I still need to do a bit of tweaking here to account for the fact that these tests are dependent on the `ExternalOIDCWithUIDAndExtraClaimMappings` feature gate being enabled.